scripts/prepare_fsd50k.py

```diff
@@ -0,0 +1,159 @@
+            genre_tracks = tracks[(tracks['track', 'genre_top'] == genre) & (tracks['set', 'subset'] == 'small')]
+            # Get ntracks_genre from the current genre
+            train_tracks = genre_tracks[:self.ntracks_genre]
+            for track_id, track in train_tracks.iterrows():
+                # get track name by id
+                subdir = f'{track_id:06}'[:3]
+                fma_track_path = os.path.join(self.dataset_home, self.dataset_name, subdir, f'{track_id:06}.mp3')
+                # Based on prob decide test vs. train split
+                save_dir = None
+                # Define DCASE path format variables
+                super_class = "music"
+                fold = "train" if np.random.rand() < self.split_prob else "test"
+                class_name = str(genre)
+                dcase_path = os.path.join(super_class, fold, str(genre), f'{track_id:06}.mp3')
+                # generate dcase to fma map
+                if fold not in self.fma_to_dcase_dict:
+                    self.fma_to_dcase_dict[fold] = {}
+                self.fma_to_dcase_dict[fold][dcase_path] = fma_track_path
+    
+
+class FSD50KDataLoad:
+    def __init__(self,
+                dataset_name="fsd50k",dataset_home=None,
+                download_dataset=False,music_dataset=None,
+                music_home=None,download_music=False,
+                dcase_sound_events_txt="../dcase_datagen/metadata/fsd50k/sound_event_fsd50k_filenames.txt",
+                metadata_path="../dcase_datagen/metadata/fsd50k"):
+        self.dataset_name = dataset_name
+        self.dataset_home = dataset_home
+        self.download_dataset = download_dataset
+        self.music = music_dataset
+        self.music_home = music_home
+        self.download_music = download_music
+        self.dcase_sound_events_txt = dcase_sound_events_txt
+        self.metadata_path = metadata_path
+        self.url_fsd_selected_txt = "https://zenodo.org/record/6406873/files/FSD50K_selected.txt"
+        self.fsd_to_dcase_dict = {} # Dict containing FSD50K to DCASE file mappings
+
+    def download_dataset(self):
+        self.fsd50k.download()
+
+    def load_dataset(self):
+        assert self.dataset_home is not None or not self.download \
+        , "Dataset home path must be provided when download is enabled."
+        self.fsd50k = soundata.initialize(self.dataset_name, self.dataset_home)
+        if self.download_dataset:
+            self.download_dataset() # download fsd50k
+        if self.music == "fma_small" and self.download_music:
+            self.music = FMADataLoad(dataset_home=self.music_home) # download fma_small
+            self.music.load_dataset()
+            self.music.gen_dataset_splits()
+            self.music.save_fma_to_dcase_json()
+        self.gen_fsd_to_dcase_dict()
+
+    def save_fsd_to_dcase_json(self):
+        with open(os.path.join(self.metadata_path, 'fsd_to_dcase.json'), 'w') as json_file:
```

Comment:
this line gave me the following error:

```
  File "/home/iran/SELD-data-generator/scripts/prepare_fsd50k.py", line 116, in  
save_fsd_to_dcase_json                                                           
    with open(os.path.join(self.metadata_path, 'fsd_to_dcase.json'), 'w') as jso 
n_file:                                                                          
FileNotFoundError: [Errno 2] No such file or directory: '../dcase_datagen/metada 
ta/fsd50k/fsd_to_dcase.json'               
```

Using absolute instead of relative paths, and informing the user about which ones need to be provided will solve this.